app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusHandlerThread.java

```diff
@@ -391,14 +391,15 @@ public boolean processRealtimeStats(byte[] data) {
             sample.setRawHPlusHealthData(record.getRawData());
             sample.setProvider(provider);
 
+            provider.addGBActivitySample(sample);
+
             sample.setSteps(sample.getSteps() - prevRealTimeRecord.steps);
```

Comment:
They are set  in line 389, stored to the DB and then the sample steps are adjusted and sent to the intent. The reason is that we store the absolute number of steps (Important for the current day steps chart), but send the number of steps since the last update to the intent (for the real time data)